drivers/spi/spi_sam.c

```diff
@@ -125,10 +125,11 @@ static int spi_sam_configure(const struct device *dev,
 	}
 
 	/* Set master mode, disable mode fault detection, set fixed peripheral
-	 * select mode.
+	 * select mode, always use the 0 peripheral as we use software gpio chip
+	 * selects.
 	 */
 	spi_mr |= (SPI_MR_MSTR | SPI_MR_MODFDIS);
-	spi_mr |= SPI_MR_PCS(spi_slave_to_mr_pcs(config->slave));
+	spi_mr |= SPI_MR_PCS(0);
```

Comment:
I have used the HW gpio chip selects, why is this restriction forced here?

Why not update the device tree to only have an I2C child at `reg 0`?

Reply:
Add the CS pins to the pinmux definition and leave out `cs-gpios`.

Reply:
Ah ok I see what you mean now, that too works

I really don't love the entire cs-gpios/reg as the chip select mechanism I have to say. Beyond confusing.

Reply:
It is a bit confusing, but it's nice to have the option to either use HW/SW CS gpios.

Reply:
Dropped this patch